python_modules/dagster/dagster/core/log.py

```diff
@@ -81,13 +81,14 @@ def _log(self, method, orig_message, message_props):
         # )
 
         for logger in self.loggers:
-            logger_method = check.is_callable(getattr(logger, method))
-            if logger.name == DAGSTER_DEFAULT_LOGGER:
-                logger_method(
-                    msg_with_multiline_structured_props, extra={DAGSTER_META_KEY: all_props}
-                )
-            else:
-                logger_method(msg_with_structured_props, extra={DAGSTER_META_KEY: all_props})
+            if hasattr(logger, method):
```

Comment:
I'm concerned about failing silently if we supply the wrong method here

Reply:
Yeah, I think this is exposing issues we currently face to some degree but haven't yet considered.

If I configure several loggers (say, a verbose logger or custom logger with my own log levels, and also something out of the box) one of which has the `.notice` method defined, and one of which doesn't, should we fail at runtime when `context.log.notice()` is called, or should we call `.notice` on the loggers that have it defined? (And what should we do if we accidentally call `.notise()`)

Reply:
ah i see. that makes sense. 

I think if the user is using multiple loggers and they have different log levels we should force them to declare what the logging interface they want is at some level. e.g. they should have to say that if they log to log.some_level how that translates to console logging

Reply:
ok, i think i might have a solution and will throw it up. it makes implementing a logger that defines new log levels a little trickier, but that's ok.